cmd/apply.go

```diff
@@ -1,18 +1,72 @@
 package cmd
 
 import (
+	"bytes"
+	"encoding/json"
 	"fmt"
+	"io"
+	"net/http"
 
 	"github.com/spf13/cobra"
 )
 
-var applyCmd = &cobra.Command{
-	Use:   "apply",
-	Short: "Apply the stack file",
-	Long:  `apply applies the stack file.`,
-	Args:  cobra.MinimumNArgs(1),
-	RunE: func(cmd *cobra.Command, args []string) error {
-		fmt.Printf("hello %v\n", args)
-		return nil
-	},
+// TODO: Auto generate these values from Brain API
+type NewDeploymentRequest struct {
+	EnvironmentName string `json:"environmentName"`
+}
+
+type NewDeploymentResponse struct {
+	DeploymentId string `json:"deploymentId"`
+	SessionToken string `json:"sessionToken"`
+	UploadUrl    string `json:"uploadUrl"`
+}
+
+var (
+	env string
+
+	applyCmd = &cobra.Command{
+		Use:   "apply",
+		Short: "Apply the stack file",
+		Long:  `apply applies the stack file.`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Do we only want to allow `test`, `stage` & `prod` values here or can they be named anything?
+			env, err := cmd.Flags().GetString("env")
+
+			if err != nil {
+				return err
+			}
+
+			deploymentReq := new(NewDeploymentRequest)
+			deploymentReq.EnvironmentName = env
+			body, err := json.Marshal(deploymentReq)
+
+			if err != nil {
+				return err
+			}
+
+			res, err := http.Post("http://localhost:8080/api/cli/deployment", "application/json", bytes.NewBuffer(body))
```

Comment:
Would it be better for me to abstract all the http client stuff into its own file?